File: 48-rotate-image/rotate-image.java
class Solution{
    public void rotate(int[][] mat){
        int n=mat.length;
    
        for(int i=0; i<n; i++){
            for(int j=i; j<n; j++){
                int t=mat[i][j];
                mat[i][j]=mat[j][i];
                mat[j][i]=t;
            }
        }

        for(int i=0; i<n; i++){
            int left=0, right=n-1;
            while(left<right){
                int t=mat[i][left];
                mat[i][left]=mat[i][right];
                mat[i][right]=t;
                left++;
                right--;
            }
        }
    }
}

File: Medium/48-rotate-image/rotate-image.py
class Solution:
    def rotate(self, mat: List[List[int]]) -> None:
        n=len(mat)

        for i in range(n):
            for j in range(i, n):
                mat[i][j], mat[j][i] = mat[j][i], mat[i][j]

        for i in range(n):
            left, right = 0, n-1
            while left < right:
                mat[i][left], mat[i][right] = mat[i][right], mat[i][left]
                left += 1
                right -= 1
        
